core/src/main/java/hudson/slaves/NodeProvisioner.java

```diff
@@ -327,6 +331,9 @@ public void run() {
         } finally {
             provisioningLock.unlock();
         }
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.finer(() -> "ran update on " + label + " in " + (System.nanoTime() - start) / 1_000_000 + "ms");
```

Comment:
I thought the whole point of the `MessageSupplier` was that it was not invoked unless the logging was enabled thus should be safe without the calling wrap?  (like you have above in [line182](https://github.com/jenkinsci/jenkins/pull/4081/files#diff-0558cec054efaecfd6bbee9cbbccdab4R182)  asking for my education as I guess there is a reason you did it.

Reply:
Merely that my IDE hint to the effect that you should use either a message template or a lambda is not clever enough to notice that the statement is gated inside `isLoggable`. Happy to delete `() ->` if it is confusing you.